### DISCUSSION TITLE
Tracking upstream dependency updates for Linux kernel

### DISCUSSION THREAD

**Discussion by samip5** [renovatebot/renovate#25615] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I would like to track a variable https://github.com/vyos/vyos-build/blob/sagitta/data/defaults.toml#L17 in my Github workflow that builds a vyos custom image so that when it's updated in upstream vyos-build repository, I would also get an auto-update for it.

### What you tried so far.

I have tried to source the information from docs, if it's possible but I'm not sure from them if it is currently?

**Comment by rarkins** [comment]
This would be a job for the [custom datasource](https://docs.renovatebot.com/modules/datasource/custom/) however today it only supports JSON and plain . @secustor is it still possible to take a TOML remote result and regex parse it as plain?

**Reply by secustor** [reply]
No, currently not possible. 

Tough should be possible with a new `regex` custom datasource: 

Where we parse a http response body text using regex and then each occurence will be a new version object. 
Named groups would be translated into fields and after that jsonata rules would be applied.

**Reply by rarkins** [reply]
Would you be able to create a feature request issue for that? I'm not sure I'll capture it correctly

**Reply by samip5** [reply]
You, aka who are you talking to @rarkins?

**Reply by rarkins** [reply]
The question was to @secustor

**Reply by secustor** [reply]
Will do some time today.

**Reply by viceice** [reply]
we should probably add support for toml and yaml. should be pretty easy to reuse JSON datasource.

**Reply by secustor** [reply, marked as answer]
- https://github.com/renovatebot/renovate/issues/25636
